internal/consul/version.go

```diff
@@ -37,3 +38,33 @@ func IsValidVersion(version string, versions []*consulmodels.HashicorpCloudConsu
 func NormalizeVersion(version string) string {
 	return "v" + strings.TrimPrefix(version, "v")
 }
+
+// VersionsToString converts a slice of version pointers to a string of their comma delimited values.
+func VersionsToString(versions []*consulmodels.HashicorpCloudConsul20200826Version) string {
+	var recommendedVersion string
+	var otherVersions []string
+
+	for _, v := range versions {
+		if v == nil {
+			continue
+		}
+
+		if v.Status == consulmodels.HashicorpCloudConsul20200826VersionStatusRECOMMENDED {
+			recommendedVersion = v.Version
+		} else {
+			otherVersions = append(otherVersions, v.Version)
+		}
+	}
+
+	// No other versions found, return recommended even if it's empty
+	if len(otherVersions) == 0 {
+		return recommendedVersion
+	}
+
+	// No recommended found, return others
+	if recommendedVersion == "" {
+		return strings.Join(otherVersions, ", ")
+	}
+
+	return fmt.Sprintf("%s, %s", recommendedVersion, strings.Join(otherVersions, ", "))
```

Comment:
Would it be weird to prepend this string with something like `recommended: ` (and same for line 61)? It might be overkill, but I think there's an edge case where eg. `v1.9.0, v1.8.6, v1.8.4` would be produced by two configurations: no recommended version, and `v1.9.0` recommended version.

Reply:
I can definitely get behind adding it for this line. On 61 since there would only be 1 version returned (the recommended one) it seems like overkill 🤷